utilities/packages/typescript/react-libs/src/hooks/useGenerateReport.tsx

```diff
@@ -0,0 +1,41 @@
+// A small custom hook that manages the react-query/api calls for the "Export" feature. 
+import { useMutation } from "@tanstack/react-query"
+
+import { generateReport } from "../queries"
+import { ItemSubType } from "../provena-interfaces/RegistryModels"
+
+export interface useGenerateReportProps {
+    // What is the Node ID (either model run or study entity)
+    id: string | undefined
+    itemSubType: ItemSubType | undefined
+    // Chosen depth by the user.
+    depth: number
+}
+
+export const useGenerateReport = (props: useGenerateReportProps) => {
+
+    // Validate that everything is present correctly.
+    const requiredItems = [props.id, props.itemSubType, props.depth]
+    const dataReady = !requiredItems.some((i) => {
+        return i === undefined;
+    });
+
+    // Setting up the useQuery
+    const generateReportQuery = useMutation({
+        mutationFn: () => {
+            return generateReport({
+                id: props.id!!,
+                item_subtype: props.itemSubType!!,
+                depth: props.depth
+
+            })
+        }
+    })
```

Comment:
I usually return undefined for the mutation if the data is not ready so that the user needs to explicitly think about that possibility. So here

```

Reply:
I'm not sure what you mean, this should work??

```
const query = useMutate(...); 
const dataReady = ...;

return {dataReady: dataReady, mutate: dataReady ? query : undefined}
```

Reply:
Sorry, I should have explained that more clearly. This approach is fine, but it does not allow us to further de-structure all the other attributes available within useMutation directly. 

E.g you can't do `const {mutate, isError, isLoading, isSuccess} = useGenerateReport({....})` , but that's okay as the attributes can still be accessed by the mutate object and make sense this way as the attributes should only be accessed after dataReady is true.  

I have updated it as you have suggested.